compiler/vm/vm.nim

```diff
@@ -222,28 +251,40 @@ proc createStrKeepNode(x: var TFullReg; keepNode=true) =
     # It not only hackey, it is also wrong for tgentemplate. The primary
     # cause of bugs like these is that the VM does not properly distinguish
     # between variable definitions (var foo = e) and variable updates (foo = e).
+]#
 
 include vmhooks
 
+# Utility accessors
+
+func intVal(x: TFullReg): BiggestInt {.inline.} = x.atomVal.intVal
+func `intVal=`(x: var TFullReg, val: BiggestInt) {.inline.} = x.atomVal.intVal = val
+
+func floatVal(x: TFullReg): BiggestFloat {.inline.} = x.atomVal.floatVal
+func `floatVal=`(x: var TFullReg, val: BiggestFloat) {.inline.} = x.atomVal.floatVal = val
+
+
 template createStr(x) =
   x.node = newNode(nkStrLit)
 
 template createSet(x) =
   x.node = newNode(nkCurly)
 
 proc moveConst(x: var TFullReg, y: TFullReg) =
+  # XXX what is this proc meant to do?
```

Comment:
Move operation, as a guess, if a register holds AST for a const expression, then we can evaluate it. After evaluation we can move the computed value in place of the AST expression with the value used for subsequent operations.